components/scream/src/physics/shoc/shoc_assumed_pdf_impl.hpp

```diff
@@ -0,0 +1,322 @@
+        const auto tsign = ekat::abs(tmp_val_1 - tmp_val_2);
+        Skew_thl.set(tsign>sp(0.4), sp(1.2)*Skew_w);
+        Skew_thl.set(tsign>sp(0.2) && tsign<=sp(0.4), (((sp(1.2)*Skew_w)/sp(0.2))*(tsign-sp(0.2))));
+      }
+
+      thl2_1.set(condition,
+                 ekat::min(100,
+                           ekat::max(0, (3*tmp_val_1*(1 - a*ekat::square(tmp_val_2) - (1-a)*ekat::square(tmp_val_1))
+                                         - (Skew_thl - a*ekat::cube(tmp_val_2) - (1 - a)*ekat::cube(tmp_val_1)))
+                                         /(3*a*(tmp_val_1 - tmp_val_2))*thlsec)));
+      thl2_2.set(condition,
+                 ekat::min(100,
+                           ekat::max(0, (-3*tmp_val_2*(1 - a*ekat::square(tmp_val_2)
+                                         - (1 - a)*ekat::square(tmp_val_1))
+                                         + (Skew_thl - a*ekat::cube(tmp_val_2) - (1 - a)*ekat::cube(tmp_val_1)))
+                                         /(3*(1 - a)*(tmp_val_1 - tmp_val_2))*thlsec)));
+
+      thl1_1.set(condition, tmp_val_2*sqrtthl+thl_first);
+      thl1_2.set(condition, tmp_val_1*sqrtthl+thl_first);
+
+      sqrtthl2_1.set(condition, ekat::sqrt(thl2_1));
+      sqrtthl2_2.set(condition, ekat::sqrt(thl2_2));
+    }
+
+    // Find parameters for total water mixing ratio
+    Spack qw1_1(qw_first), qw1_2(qw_first), qw2_1(0), qw2_2(0),
+          sqrtqw2_1(0), sqrtqw2_2(0);
+    {
+      const Smask condition = qwsec > (rt_tol*rt_tol) && ekat::abs(w1_2 - w1_1) > w_thresh;
+
+      const Spack corrtest2 = ekat::max(-1, ekat::min(1, wqwsec/(sqrtw2*sqrtqt)));
+      const Spack tmp_val_1(-corrtest2/w1_1), tmp_val_2(-corrtest2/w1_2);
+
+      const auto tsign = ekat::abs(tmp_val_1 - tmp_val_2);
+      Spack Skew_qw(0);
+      Skew_qw.set(tsign>sp(0.4), sp(1.2)*Skew_w);
+      Skew_qw.set(tsign>sp(0.2) && tsign<=sp(0.4), (((sp(1.2)*Skew_w)/sp(0.2))*(tsign-sp(0.2))));
+
+      qw2_1.set(condition,
+                ekat::min(100,
+                          ekat::max(0, (3*tmp_val_1*(1 - a*ekat::square(tmp_val_2) - (1 - a)*ekat::square(tmp_val_1))
+                                        - (Skew_qw - a*ekat::cube(tmp_val_2) - (1 - a)*ekat::cube(tmp_val_1)))
+                                        /(3*a*(tmp_val_1 - tmp_val_2))))*qwsec);
+      qw2_2.set(condition,
+                ekat::min(100,
+                          ekat::max(0, (-3*tmp_val_2*(1 - a*ekat::square(tmp_val_2) - (1 - a)*ekat::square(tmp_val_1))
+                                        + (Skew_qw - a*ekat::cube(tmp_val_2) - (1 - a)*ekat::cube(tmp_val_1)))
+                                        /(3*(1 - a)*(tmp_val_1 - tmp_val_2))))*qwsec);
+
+      qw1_1.set(condition, tmp_val_2*sqrtqt+qw_first);
+      qw1_2.set(condition, tmp_val_1*sqrtqt+qw_first);
+
+      sqrtqw2_1.set(condition, ekat::sqrt(qw2_1));
+      sqrtqw2_2.set(condition, ekat::sqrt(qw2_2));
+    }
+
+    // Convert from tilda variables to "real" variables
```

Comment:
"tilde" variables

Reply:
Hmm. I see the original spelling. Either they're referencing some math by some person named Tilda, or they just don't know how to spell "tilde" (like the squiggle over certain math symbols).

Reply:
My guess is it should be "tilde"..

Reply:
Should I change the function name in shoc.F90?

Reply:
🤷‍♂️  sure!